packages/utils/typescript/lib/generators/utils.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+const ts = require('typescript');
+const prettier = require('prettier');
+const fse = require('fs-extra');
+const path = require('path');
+const chalk = require('chalk');
+const assert = require('assert');
+
+const { factory } = ts;
+
+/**
+ * Aggregate the given TypeScript nodes into a single string
+ *
+ * @param {ts.Node[]} definitions
+ * @return {string}
+ */
+const emitDefinitions = (definitions) => {
+  const nodeArray = factory.createNodeArray(definitions);
+
+  const sourceFile = ts.createSourceFile(
+    'placeholder.ts',
+    '',
+    ts.ScriptTarget.ESNext,
+    true,
+    ts.ScriptKind.TS
+  );
+
+  const printer = ts.createPrinter({ omitTrailingSemicolon: true });
+
+  return printer.printList(ts.ListFormat.MultiLine, nodeArray, sourceFile);
+};
+
+/**
+ * Save the given string representation of TS nodes in a file
+ * If the given directory doesn't exist, it'll be created automatically
+ *
+ * @param {string} dir
+ * @param {string} file
+ * @param {string} content
+ *
+ * @return {Promise<string>} The path of the created file
+ */
+const saveDefinitionToFileSystem = async (dir, file, content) => {
+  const filepath = path.join(dir, file);
+
+  fse.ensureDirSync(dir);
+  await fse.writeFile(filepath, content);
+
+  return filepath;
+};
+
+/**
+ * Format the given definitions.
+ * Uses the existing config if one is defined in the project.
+ *
+ * @param {string} content
+ * @returns {Promise<string>}
+ */
+const format = async (content) => {
+  const configFile = await prettier.resolveConfigFile();
+  const config = configFile
+    ? await prettier.resolveConfig(configFile)
+    : // Default config
+      {
+        singleQuote: true,
+        useTabs: false,
+        tabWidth: 2,
+      };
+
+  Object.assign(config, { parser: 'typescript' });
```

Comment:
Should we start with the default config and merge in their config file so that if they only define, for example, tabWidth we still use our default singleQuote and useTabs? 

Otherwise someone could create an empty config file and suddenly everything changes. But it might also be bad to add values to their config. What do you all think?

Reply:
I think if someone wants to use a custom config, we should make it pure and don't mutate it with our changes, otherwise it wouldn't be consistent nor react the same ways as tutoriels or examples people could be following & such.